examples/plugins/org.jboss.tools.project.examples/src/org/jboss/tools/project/examples/internal/ProjectExamplesActivator.java

```diff
@@ -505,9 +506,19 @@ public void run() {
 										}
 										if (files.length > 0) {
 											try {
-												if (txtFile) {
+												IEditorDescriptor descriptor = IDE.getDefaultEditor(files[0], true);
+												if (descriptor != null && !"org.eclipse.ui.systemExternalEditor".equals(descriptor.getId())) { //$NON-NLS-1$
+													IDE.openEditor(page, files[0], descriptor.getId());
+												} else if (txtFile) {
 													IFileEditorInput input = new FileEditorInput(files[0]);
 													IDE.openEditor(page, input, EditorsUI.DEFAULT_TEXT_EDITOR_ID);
+												} else if (mdFile) {
+													IFileEditorInput input = new FileEditorInput(files[0]);
+													try {
+														IDE.openEditor(page, input, "org.eclipse.mylyn.wikitext.ui.editor.markupEditor"); //$NON-NLS-1$

```

Comment:
Isn't the API smart enough to infer the proper editor depending on the file type?